tm2/pkg/p2p/config/config.go

```diff
@@ -36,10 +26,10 @@ type P2PConfig struct {
 	UPNP bool `json:"upnp" toml:"upnp" comment:"UPNP port forwarding"`
 
 	// Maximum number of inbound peers
-	MaxNumInboundPeers int `json:"max_num_inbound_peers" toml:"max_num_inbound_peers" comment:"Maximum number of inbound peers"`
+	MaxNumInboundPeers uint64 `json:"max_num_inbound_peers" toml:"max_num_inbound_peers" comment:"Maximum number of inbound peers"`
```

Comment:
Unsigned may be better than signed if we don't need to use patterns like `-1`. However, since we're constrained by the OS limits, a size of 64 may be excessive.